### ISSUE TITLE
Multiple 'sentry-cli releases deploys $VERSION new -e env' get overwritten by the last one resulting in broken releases.

### ISSUE THREAD

**Issue by maximLyakhov** [getsentry/sentry-cli#1900]
### Environment

OS: darwin arm64
Node: 20.10.0
Package Manager: pnpm 8.14.0
@sentry/cli: 2.25.0

### Steps to Reproduce

    sentry-cli --url https://sentry.myorg.dev/ login --auth-token $SENTRY_AUTH_TOKEN
    sentry-cli releases new $VERSION
    sentry-cli releases set-commits $VERSION --auto --ignore-missing
    sentry-cli sourcemaps inject ./dist/app/stage -r=$VERSION
    sentry-cli sourcemaps inject ./dist/app/prod -r=$VERSION
    sentry-cli sourcemaps upload ./dist/app/stage -r=$VERSION --dist=stage
    sentry-cli sourcemaps upload ./dist/app/prod -r=$VERSION --dist=prod
    sentry-cli releases deploys $VERSION new -e stage -r=$VERSION -n=$VERSION
    sentry-cli releases deploys $VERSION new -e prod -r=$VERSION -n=$VERSION
    sentry-cli releases finalize $VERSION

### Expected Result

I expect sentry-cli to create two releases for "stage" and "prod" environments.

### Actual Result

I get notified on email with two messages

    Version app@hash was deployed to stage
    Version app@hash was deployed to prod

Sentry web interface shows that I have "version app-name date | prod" release
and release containing only "version".

Clicking on this "empty" release > Source Maps [0 artifacts] >

    There are no artifacts in this archive.

Clicking on "valid" release > Source Maps [148 artifacts] > 

    Artifact Bundles
    BUNDLE ID            | ARTIFACTS | DATE MODIFIED | DATE UPLOADED
    Bundle ID 
    1 Release associated |        74 |          date | date
    Bundle ID 
    1 Release associated |        74 |          date | date

    Release Bundles
    NAME     | ARTIFACTS    | DATE UPLOADED
    app@hash | 0 (no value)

I guess something is messing up by double deploy.
I don't really need multiple environments since code on stage and prod is identical except api addresses,
but `--env` is required for `sentry-cli releases deploys new`.

Also I would like to know why my release bundles is empty even on "valid" release.
And how to upload sourcemaps injected with debug ids without deploying them with actual app.

Please point me where I've mistaken.

**Comment by szokeasaurusrex** [comment]
Hi @maximLyakhov, thank you for reaching out. I will investigate this

**Comment by szokeasaurusrex** [comment]
@maximLyakhov, can you please try again, but without passing a `--dist` argument? I think omitting the `--dist` could solve your problem

**Comment by maximLyakhov** [comment]
@szokeasaurusrex thank you for your time. Result is the same without `--dist`.

**Comment by szokeasaurusrex** [comment]
@maximLyakhov, okay, I will investigate further. If you are using Sentry SaaS, would you be able to share the link to the page in Sentry where you see the "There are no artifacts in this archive" message?

**Comment by maximLyakhov** [comment]
@szokeasaurusrex I think we use self hosted `sentry.<our-org>.dev`.

**Comment by szokeasaurusrex** [comment]
Yeah, SaaS would be if you are using the sentry.io. Thank you for the screenshot, though!

**Comment by maximLyakhov** [comment]
Also some informational screenshots:
<img width="1220" alt="Screenshot 2024-01-19 at 19 28 48" src="https://github.com/getsentry/sentry-cli/assets/75264285/457195f0-4241-4fe7-a606-790bba557d2f">
<img width="1220" alt="Screenshot 2024-01-19 at 19 29 03" src="https://github.com/getsentry/sentry-cli/assets/75264285/d1544e1d-1774-44c6-ac06-d4873bddae1c">

**Comment by maximLyakhov** [comment]
What do I do wrong eventually?

**Comment by szokeasaurusrex** [comment]
@maximLyakhov sorry, this issue fell through the cracks. I looked into your issue again, and I tried to reproduce the problem by following the commands you provided.

As far as I can tell, however, the behavior you have described appears to be expected. The reason your release bundles appear to be empty is because they in fact are empty; since you are using the `sourcemaps inject` command, you are uploading "Artifact Bundles" instead of "Release Bundles". If you check the "Artifact Bundles" tab on the Source Maps page in Sentry, you should see the uploaded sourcemaps, and they should be correctly associated with the provided release, as in the screenshot below:
<img width="1552" alt="image" src="https://github.com/getsentry/sentry-cli/assets/7881302/ed6f29fe-02bd-4e32-aed2-ba146d984d5e">

Are you able to see the Artifact Bundle in Sentry? And, just to clarify, is the only problem you are experiencing that the Release Bundle is empty (which is the expected behavior)? Or, are you observing any other problems, such as your error events not being symbolicated properly?

**Comment by maximLyakhov** [comment]
@szokeasaurusrex thank you! What should I call instead of `sourcemaps inject`? 
Can you show me your commands resulting in valid release?

**Comment by szokeasaurusrex** [comment]
@maximLyakhov I will answer each of your questions individually below:


> Are Release Bundles deprecated? Why do they show up?

Release Bundles are generated when you use our [legacy sourcemap uploading methods](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/legacy-uploading-methods/?original_referrer=https%3A%2F%2Fwww.google.com%2F). I don't believe that Release Bundles are officially "deprecated" because they are still being kept around for people who are using older versions of the Sentry SDKs or CLI. However, if you are able to, we recommend using Artifact Bundles instead.



> What should I call instead of `sourcemaps inject`?

You should be using `sourcemaps inject`. `sourcemaps inject` adds Debug IDs to your minified JS files and sourcemap files in order to associate the minified files with the sourcemaps. By using `sourcemaps inject`, you are able to take advantage of Artifact Bundles.



> Can you show me your commands resulting in valid release?

The commands you are using should already result in a valid release. As long as you see the Artifact Bundles in Sentry, and as long as the correct releases are displayed under "Associated Releases," everything should be good!

_______________

In short, it seems like you are already doing everything correctly! You can just ignore the Release Bundles page, since it would only be applicable to you if you were using the legacy sourcemap uploading method.

For more information about Artifact Bundles, please see [this docs page](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/artifact-bundles/), which explains what Artifact Bundles are and how the compare to the older Release Bundles.

**Comment by szokeasaurusrex** [comment]
Awesome, glad I could assist!
